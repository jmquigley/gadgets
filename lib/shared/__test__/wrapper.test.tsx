'use strict';

test('Test creation of the ErrorHandler', () => {
	// TODO: add ErrorHandler creation test
});
