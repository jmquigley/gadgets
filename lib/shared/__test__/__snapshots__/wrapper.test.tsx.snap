// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a default Wrapper 1`] = `
<Wrapper
  className="test-class"
  onError={
    [MockFunction] {
      "calls": Array [
        Array [
          "test error",
          Object {
            "componentStack": "test stack",
          },
        ],
      ],
    }
  }
  reset={true}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "selectedBackgroundColor": "#1f8a70",
        "selectedForegroundColor": "#ffe11a",
        "tooltipBackgroundColor": "#708090",
        "tooltipForegroundColor": "#ffffff",
        "transitionDelay": "0.5s",
      }
    }
  >
    <styled.div
      className="ui-error test-class"
    >
      <div
        className="ui-error test-class sc-bdVaJa hfTfIP"
      >
        <span
          className="ui-error-message"
        >
          Error in component '
          '
        </span>
        <details
          className="ui-error-stack"
          style={
            Object {
              "whiteSpace": "pre-wrap",
            }
          }
        >
          test error
          <br />
          test stack
        </details>
      </div>
    </styled.div>
  </ThemeProvider>
</Wrapper>
`;

exports[`Test creation of the Wrapper props object 1`] = `
Object {
  "children": null,
  "onError": [Function],
  "reset": false,
}
`;

exports[`Test disabling the Wrapper 1`] = `
<Wrapper
  className="test-class"
  disabled={true}
  onError={[MockFunction]}
  reset={false}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "selectedBackgroundColor": "#1f8a70",
        "selectedForegroundColor": "#ffe11a",
        "tooltipBackgroundColor": "#708090",
        "tooltipForegroundColor": "#ffffff",
        "transitionDelay": "0.5s",
      }
    }
  >
    <h1>
      test header
    </h1>
  </ThemeProvider>
</Wrapper>
`;
